.github/workflows/build-runtime-watcher.yml

```
@@ -0,0 +1,35 @@
+name: build-runtime-watcher
+
+on:
+  push:
+    branches:
+      - main  # This will get tagged with `latest` and `v{{DATE}}-{{COMMIT_HASH_SHORT}}`
+  pull_request_target:
+    types: [ opened, edited, synchronize, reopened, ready_for_review ]
```

```suggestion
  push:
    branches:
  pull_request:
    branches:
      - main
```

You dont need to list all types of PRs. I would suggest just specifying the branch, like we do in our other actions

---

This has been implemented as per the guide [listed here](https://github.com/kyma-project/test-infra/blob/main/cmd/image-builder/README.md#quickstart-guide).

It has to stay as `pull_request_target` otherwise the PR won't have image builds on every push while still in work-in-progress mode. Moreover, that's the event which triggers the necessary actions from `kyma-project/test-infra/.github/workflows/image-builder.yml@main`, changing it will result in pipelines failures. This is also the action which tags it with the `PR-xxx` format.

The reason for mentioning types under `pull_request_target` is because not all types are default ones, so it needed to have extra + the default ones. It has been tried and tested with the `template-operator` manually and autonomously.

For the `push.branches.main`, the `push` event triggers image creation with `latest` and `v{{DATE}}-{{COMMIT_HASH_SHORT}}` tags.